packages/core/src/saml-applications/routes/utils.ts

```diff
@@ -28,8 +29,366 @@ import {
   samlAttributeNameFormatBasic,
   samlValueXmlnsXsi,
 } from '../libraries/consts.js';
+import { buildSingleSignOnUrl, buildSamlIdentityProviderEntityId } from '../libraries/utils.js';
 import { type SamlApplicationDetails } from '../queries/index.js';
 
+type ValidSamlApplicationDetails = {
+  secret: string;
+  entityId: string;
+  acsUrl: SamlAcsUrl;
+  redirectUri: string;
+  privateKey: string;
+  certificate: string;
+};
+
+export class SamlApplication {
+  public static validateSamlApplicationDetails = (
+    details: SamlApplicationDetails
+  ): ValidSamlApplicationDetails => {
+    const {
+      entityId,
+      acsUrl,
+      oidcClientMetadata: { redirectUris },
+      privateKey,
+      certificate,
+      secret,
+    } = details;
+
+    assertThat(acsUrl, 'application.saml.acs_url_required');
+    assertThat(entityId, 'application.saml.entity_id_required');
+    assertThat(redirectUris[0], 'oidc.invalid_redirect_uri');
+
+    assertThat(privateKey, 'application.saml.private_key_required');
+    assertThat(certificate, 'application.saml.certificate_required');
+
+    return {
+      secret,
+      entityId,
+      acsUrl,
+      redirectUri: redirectUris[0],
+      privateKey,
+      certificate,
+    };
+  };
+
+  protected static getSamlIdpAndSp = ({
```

Comment:
Better split into multiple methods:
 - buildSamlIdentityProvider
 - buildSamlServiceProvider
 - buildLoginResponseTemplate

Reply:
nit: You can define these utility methods separately in this file. Since they are only used once in the constructor, making them static methods is unnecessary.